src/main/frontend/components/repo.cljs

```diff
@@ -198,26 +204,18 @@
                               (db/get-repo-path repo)))
             repos (state/sub [:me :repos])
             repos (remove (fn [r] (= config/local-repo (:url r))) repos)
-            switch-repos (remove (fn [repo] (= current-repo (:url repo)))
+            switch-repos (remove (fn [repo]
+                                   (= current-repo (:url repo)))
                                  repos)
             repo-links (mapv
                         (fn [{:keys [id url]}]
                           (let [repo-path (get-repo-name url)
-                                short-repo-name (shorten-repo-name repo-path)]
-
-                            {:title        [:span short-repo-name]
-                             :hover-detail repo-path ;; show full path on hover
-                             :options      {:on-click
-                                            (fn []
-                                              (repo-handler/push-if-auto-enabled! (state/get-current-repo))
-                                              (state/set-current-repo! url)
-                                                 ;; load config
-                                              (common-handler/reset-config! url nil)
-                                              (shortcut/refresh!)
-                                              (when-not (= :draw (state/get-current-route))
-                                                (route-handler/redirect-to-home!))
-                                              (when-let [dir-name (config/get-repo-dir url)]
-                                                (fs/watch-dir! dir-name)))}}))
+                                short-repo-name (text/get-graph-name-from-path repo-path)]
```

Comment:
Since `text/get-graph-name-from-path` is used in the "All Graph" page, we can reuse it here.